databricks_cli/stack/api.py

```diff
@@ -0,0 +1,353 @@
+        template.
+
+        :param resource: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE and
+        RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        :return: dict resource_deploy_info- A dictionary of deployment information of the
+        resource to be stored at deploy time. It includes the resource id of the resource along
+        with the physical id and deploy output of the resource.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {'job_id': 123},
+        'timestamp': 123456789, 'deploy_output': {..}}
+        """
+        try:
+            resource_id = resource[RESOURCE_ID]
+            resource_service = resource[RESOURCE_SERVICE]
+            resource_properties = resource[RESOURCE_PROPERTIES]
+        except KeyError as e:
+            raise StackError("%s doesn't exist in resource config" % str(e))
+
+        # Deployment
+        physical_id = self._get_deployed_resource_physical_id(resource_id, resource_service)
+
+        if resource_service == JOBS_SERVICE:
+            physical_id, deploy_output = self.deploy_job(resource_id, resource_properties,
+                                                         physical_id)
+        else:
+            raise StackError("Resource service '%s' not supported" % resource_service)
+
+        resource_deploy_info = {RESOURCE_ID: resource_id, RESOURCE_SERVICE: resource_service,
+                                RESOURCE_DEPLOY_TIMESTAMP: datetime.now(),
+                                RESOURCE_PHYSICAL_ID: physical_id,
+                                RESOURCE_DEPLOY_OUTPUT: deploy_output}
+        return resource_deploy_info
+
+    def deploy(self, config_path):  # overwrite to be added
+        """
+        Deploys a stack given stack JSON configuration template at path config_path.
+
+        Loads the JSON template as well as status JSON if stack has been deployed before.
+        After going through each of the resources and deploying them, stores status JSON
+        of deployment with deploy status of each resource deployment.
+
+        :param config_path: Path to stack JSON configuration template. Must have the fields of
+        'name', the name of the stack and 'resources', a list of stack resources.
+        :return: None.
+        """
+        config_dir = os.path.dirname(os.path.abspath(config_path))
+        cli_cwd = os.getcwd()
+        os.chdir(config_dir)  # Switch current working directory to where json config is stored
+        try:
+            parsed_conf = self._parse_config_file(config_path)
+            if STACK_NAME not in parsed_conf:
+                raise StackError("'%s' not in configuration" % STACK_NAME)
+            stack_name = parsed_conf[STACK_NAME]
+            status_path = self._generate_stack_status_path(config_path)
+            self._load_stack_status(status_path)
```

Comment:
Delegate this functionality to the cli.py. You can have your API client expose the contents of the stack status. The caller should can decide what to do with it.